pkg/cloudprovider/providers/azure/azure_instances.go

```diff
@@ -126,6 +126,44 @@ func (az *Cloud) InstanceID(name types.NodeName) (string, error) {
 			}
 		}
 	}
+
+	if az.Config.ScaleMode == scaleModeVMSS {
+		id, err := az.getVmssInstanceID(name)
+		if err == cloudprovider.InstanceNotFound || err == ErrorNotVmssInstance {
+			// Retry with vmas mode because master nodes may not belong to any vmss.
```

Comment:
nit s/vmas/vmss/

Reply:
nevermind...

Reply:
it is retried with vmas because master nodes may not belong to any scale sets.